nilearn/plotting/img_comparison.py

```diff
@@ -469,32 +469,19 @@ def _sanitize_masker(masker, image_type, ref_img):
         else:
             masker = SurfaceMasker()
 
-    if image_type == "volume":
-        if not isinstance(masker, (NiftiMasker, Nifti1Image, str, Path)):
-            raise TypeError(
-                "'masker' must be NiftiMasker or Niimg-Like "
-                "for volume based images.\n"
-                f"Got {type(masker)}"
-            )
-        elif isinstance(masker, (Nifti1Image, str, Path)):
-            masker = NiftiMasker(
-                mask_img=masker,
-                target_affine=ref_img.affine,
-                target_shape=ref_img.shape,
-            )
+    check_compatibility_mask_and_images(masker, ref_img)
 
-    else:
-        if not isinstance(masker, (SurfaceMasker, SurfaceImage)):
-            raise TypeError(
-                "'masker' must be SurfaceMasker or SurfaceImage "
-                "for surface based images.\n"
-                f"Got {type(masker)}"
-            )
-        if isinstance(masker, SurfaceImage):
-            check_same_n_vertices(ref_img.mesh, masker.mesh)
-            masker = SurfaceMasker(
-                mask_img=masker,
-            )
+    if isinstance(masker, NiimgLike):
+        masker = NiftiMasker(
+            mask_img=masker,
+            target_affine=ref_img.affine,
+            target_shape=ref_img.shape,
+        )
+    elif isinstance(masker, SurfaceImage):
+        check_same_n_vertices(ref_img.mesh, masker.mesh)
+        masker = SurfaceMasker(
+            mask_img=masker,
+        )
```

Comment:
I am wondering if this check can also happen within the `check_compatibility_mask_and_images` function

Reply:
maybe but I'd rather not do that just yet